Bottom: d2a0af06e7e4d0d01a41459bd8e5a12e1108e603
Top:    23ca2576e16cd4a56dcaa6a50550a95eebc7824c
Author: Andrew Cooper <andrew.cooper@novatechweb.com>
Date:   2018-05-11 16:29:54 -0500

Move ansible role vars to role defaults

This allows other parts of the playbook i.e. host vars, command line the
ability to override these defaults when necessary.


---
diff --git a/roles/docker-exim4/defaults/main.yml b/roles/docker-exim4/defaults/main.yml
index 31edfea..24ba098 100644
--- a/roles/docker-exim4/defaults/main.yml
+++ b/roles/docker-exim4/defaults/main.yml
@@ -16,3 +16,37 @@ container_port_map:
 
 # Network names
 docker_network_frontend: 'frontend'
+
+# the repository containing Dockerfile to build the image
+exim4_image_repo: https://github.com/novatechweb/docker-exim4.git
+
+# the name of the image being duilt and used for the container
+exim4_image_name: '{{ docker_registry_username }}/exim4'
+
+# the name of the container being started
+exim4_container_name: '{{ docker_name_prefix }}exim4'
+
+# the name of the data-volume used by the container
+exim4_dv_name: '{{ docker_name_prefix }}exim4_DV'
+
+# restore directories to temporarly store data being restored into docker containers
+exim4_docker_backup_dir: '{{ docker_backup_dir }}/EXIM4'
+host_exim4_docker_restore_dir: '{{ bacula_dest }}{{ exim4_docker_backup_dir }}'
+
+# files restored from tape
+exim4_backup_file: '/exim4.tar'
+
+# Data volume directory for the syslog data
+exim4_syslog_datavolume_dir: '/var/log/docker_container/{{ exim4_container_name }}'
+
+
+email_accounts:
+  wiki:
+    user: wiki
+    password: '{{ wiki_email_password }}'
+  gitlab:
+    user: gitlab
+    password: '{{ gitlab_email_password }}'
+  mantisbt:
+    user: mantis
+    password: '{{ mantisbt_email_password }}'
diff --git a/roles/docker-exim4/vars/main.yml b/roles/docker-exim4/vars/main.yml
index 20d48fc..ed97d53 100644
--- a/roles/docker-exim4/vars/main.yml
+++ b/roles/docker-exim4/vars/main.yml
@@ -1,35 +1 @@
 ---
-
-# the repository containing Dockerfile to build the image
-exim4_image_repo: https://github.com/novatechweb/docker-exim4.git
-
-# the name of the image being duilt and used for the container
-exim4_image_name: '{{ docker_registry_username }}/exim4'
-
-# the name of the container being started
-exim4_container_name: '{{ docker_name_prefix }}exim4'
-
-# the name of the data-volume used by the container
-exim4_dv_name: '{{ docker_name_prefix }}exim4_DV'
-
-# restore directories to temporarly store data being restored into docker containers
-exim4_docker_backup_dir: '{{ docker_backup_dir }}/EXIM4'
-host_exim4_docker_restore_dir: '{{ bacula_dest }}{{ exim4_docker_backup_dir }}'
-
-# files restored from tape
-exim4_backup_file: '/exim4.tar'
-
-# Data volume directory for the syslog data
-exim4_syslog_datavolume_dir: '/var/log/docker_container/{{ exim4_container_name }}'
-
-
-email_accounts:
-  wiki:
-    user: wiki
-    password: '{{ wiki_email_password }}'
-  gitlab:
-    user: gitlab
-    password: '{{ gitlab_email_password }}'
-  mantisbt:
-    user: mantis
-    password: '{{ mantisbt_email_password }}'
diff --git a/roles/docker-gitlab/defaults/main.yml b/roles/docker-gitlab/defaults/main.yml
index e4de15b..1849f72 100644
--- a/roles/docker-gitlab/defaults/main.yml
+++ b/roles/docker-gitlab/defaults/main.yml
@@ -22,3 +22,29 @@ gitlab_db_user: novatech
 
 # Network names
 docker_network_frontend: 'frontend'
+
+# the name of the image being duilt and used for the container
+gitlab_image_repo: 'https://github.com/novatechweb/docker-gitlab.git'
+gitlab_image_name: '{{ docker_registry_username }}/gitlab'
+gitlab_dv_image_name: '{{ docker_registry_username }}/gitlab'
+gitlab_db_image_name: '{{ docker_registry_username }}/postgres'
+redis_image_name: '{{ docker_registry_username }}/redis'
+
+# the name of the gitlab container
+gitlab_container_name: '{{ docker_name_prefix }}gitlab'
+
+# the name of the redis container for gitlab
+gitlab_redis_container_name: '{{ docker_name_prefix }}gitlab_redis'
+
+# the name of the postgres container for gitlab
+gitlab_db_container_name: '{{ docker_name_prefix }}gitlab_db'
+
+# the name of the data-volume used by the container
+gitlab_dv_name: '{{ docker_name_prefix }}gitlab_DV'
+
+# the name of the data-volume used by the postgres container
+gitlab_db_dv_name: '{{ docker_name_prefix }}gitlab_db_DV'
+
+# restore directories to temporarly store data being restored into docker containers
+gitlab_docker_backup_dir: '{{ docker_backup_dir }}/GIT'
+host_gitlab_docker_restore_dir: '{{ bacula_dest }}{{ gitlab_docker_backup_dir }}'
diff --git a/roles/docker-gitlab/vars/main.yml b/roles/docker-gitlab/vars/main.yml
index 7813330..ed97d53 100644
--- a/roles/docker-gitlab/vars/main.yml
+++ b/roles/docker-gitlab/vars/main.yml
@@ -1,27 +1 @@
 ---
-
-# the name of the image being duilt and used for the container
-gitlab_image_repo: 'https://github.com/novatechweb/docker-gitlab.git'
-gitlab_image_name: '{{ docker_registry_username }}/gitlab'
-gitlab_dv_image_name: '{{ docker_registry_username }}/gitlab'
-gitlab_db_image_name: '{{ docker_registry_username }}/postgres'
-redis_image_name: '{{ docker_registry_username }}/redis'
-
-# the name of the gitlab container
-gitlab_container_name: '{{ docker_name_prefix }}gitlab'
-
-# the name of the redis container for gitlab
-gitlab_redis_container_name: '{{ docker_name_prefix }}gitlab_redis'
-
-# the name of the postgres container for gitlab
-gitlab_db_container_name: '{{ docker_name_prefix }}gitlab_db'
-
-# the name of the data-volume used by the container
-gitlab_dv_name: '{{ docker_name_prefix }}gitlab_DV'
-
-# the name of the data-volume used by the postgres container
-gitlab_db_dv_name: '{{ docker_name_prefix }}gitlab_db_DV'
-
-# restore directories to temporarly store data being restored into docker containers
-gitlab_docker_backup_dir: '{{ docker_backup_dir }}/GIT'
-host_gitlab_docker_restore_dir: '{{ bacula_dest }}{{ gitlab_docker_backup_dir }}'
diff --git a/roles/docker-mantisbt/defaults/main.yml b/roles/docker-mantisbt/defaults/main.yml
index c22497f..67254e9 100644
--- a/roles/docker-mantisbt/defaults/main.yml
+++ b/roles/docker-mantisbt/defaults/main.yml
@@ -20,3 +20,31 @@ mantisbt_db_user: novatech
 
 # Network names
 docker_network_frontend: 'frontend'
+
+# the name of the image being duilt and used for the container
+mantisbt_image_repo: https://github.com/novatechweb/docker-mantisbt.git
+mantisbt_image_name: '{{ docker_registry_username }}/mantisbt'
+
+mantisbt_db_image_repo: https://github.com/novatechweb/docker-mysql-data.git
+mantisbt_db_image_name: '{{ docker_registry_username }}/mysql.dv'
+
+# the name of the container being started
+mantisbt_container_name: '{{ docker_name_prefix }}mantisbt'
+
+# the name of the container being started
+mantisbt_db_container_name: '{{ docker_name_prefix }}mantisbt_db'
+
+# the name of the data-volume used by the container
+mantisbt_dv_name: '{{ docker_name_prefix }}mantisbt_DV'
+
+# the name of the mysql data-volume used by the mantisbt container
+mantisbt_db_dv_name: '{{ docker_name_prefix }}mantisbt_db_DV'
+
+# restore directories to temporarly store data being restored into docker containers
+mantisbt_docker_backup_dir: '{{ docker_backup_dir }}/MANTISBT'
+host_mantisbt_docker_restore_dir: '{{ bacula_dest }}{{ mantisbt_docker_backup_dir }}'
+
+# files restored from tape
+mantisbt_static_backup_file: '/mantisbt.tar'
+mantisbt_database_backup_file: '/mantisbtdb.sql'
+mantisbt_database_files_table_backup_file: '/mantisbtdb-files_table.sql'
diff --git a/roles/docker-mantisbt/vars/main.yml b/roles/docker-mantisbt/vars/main.yml
index fac5a9c..ed97d53 100644
--- a/roles/docker-mantisbt/vars/main.yml
+++ b/roles/docker-mantisbt/vars/main.yml
@@ -1,29 +1 @@
 ---
-
-# the name of the image being duilt and used for the container
-mantisbt_image_repo: https://github.com/novatechweb/docker-mantisbt.git
-mantisbt_image_name: '{{ docker_registry_username }}/mantisbt'
-
-mantisbt_db_image_repo: https://github.com/novatechweb/docker-mysql-data.git
-mantisbt_db_image_name: '{{ docker_registry_username }}/mysql.dv'
-
-# the name of the container being started
-mantisbt_container_name: '{{ docker_name_prefix }}mantisbt'
-
-# the name of the container being started
-mantisbt_db_container_name: '{{ docker_name_prefix }}mantisbt_db'
-
-# the name of the data-volume used by the container
-mantisbt_dv_name: '{{ docker_name_prefix }}mantisbt_DV'
-
-# the name of the mysql data-volume used by the mantisbt container
-mantisbt_db_dv_name: '{{ docker_name_prefix }}mantisbt_db_DV'
-
-# restore directories to temporarly store data being restored into docker containers
-mantisbt_docker_backup_dir: '{{ docker_backup_dir }}/MANTISBT'
-host_mantisbt_docker_restore_dir: '{{ bacula_dest }}{{ mantisbt_docker_backup_dir }}'
-
-# files restored from tape
-mantisbt_static_backup_file: '/mantisbt.tar'
-mantisbt_database_backup_file: '/mantisbtdb.sql'
-mantisbt_database_files_table_backup_file: '/mantisbtdb-files_table.sql'
diff --git a/roles/docker-mediawiki/defaults/main.yml b/roles/docker-mediawiki/defaults/main.yml
index 603000d..2fec56c 100644
--- a/roles/docker-mediawiki/defaults/main.yml
+++ b/roles/docker-mediawiki/defaults/main.yml
@@ -20,3 +20,29 @@ wiki_db_user: novatech
 
 # Network names
 docker_network_frontend: 'frontend'
+
+wiki_image_repo: https://github.com/novatechweb/docker-mediawiki.git
+
+# the name of the image being duilt and used for the container
+wiki_image_name: '{{ docker_registry_username }}/wiki'
+wiki_db_image_name: '{{ docker_registry_username }}/mysql'
+
+# the name of the container being started
+wiki_container_name: '{{ docker_name_prefix }}wiki'
+
+# the name of the container being started
+wiki_db_container_name: '{{ docker_name_prefix }}wiki_db'
+
+# the name of the data-volume used by the container
+wiki_dv_name: '{{ docker_name_prefix }}wiki_DV'
+
+# the name of the mysql data-volume used by the mediawiki container
+wiki_db_dv_name: '{{ docker_name_prefix }}wiki_db_DV'
+
+# restore directories to temporarly store data being restored into docker containers
+wiki_docker_backup_dir: '{{ docker_backup_dir }}/WIKI'
+host_wiki_docker_restore_dir: '{{ bacula_dest }}{{ wiki_docker_backup_dir }}'
+
+# files restored from tape
+wiki_static_backup_file: '/mediawiki.tar'
+wiki_database_backup_file: '/wikidb.sql'
diff --git a/roles/docker-mediawiki/vars/main.yml b/roles/docker-mediawiki/vars/main.yml
index 86f1ac2..ed97d53 100644
--- a/roles/docker-mediawiki/vars/main.yml
+++ b/roles/docker-mediawiki/vars/main.yml
@@ -1,27 +1 @@
 ---
-
-wiki_image_repo: https://github.com/novatechweb/docker-mediawiki.git
-
-# the name of the image being duilt and used for the container
-wiki_image_name: '{{ docker_registry_username }}/wiki'
-wiki_db_image_name: '{{ docker_registry_username }}/mysql'
-
-# the name of the container being started
-wiki_container_name: '{{ docker_name_prefix }}wiki'
-
-# the name of the container being started
-wiki_db_container_name: '{{ docker_name_prefix }}wiki_db'
-
-# the name of the data-volume used by the container
-wiki_dv_name: '{{ docker_name_prefix }}wiki_DV'
-
-# the name of the mysql data-volume used by the mediawiki container
-wiki_db_dv_name: '{{ docker_name_prefix }}wiki_db_DV'
-
-# restore directories to temporarly store data being restored into docker containers
-wiki_docker_backup_dir: '{{ docker_backup_dir }}/WIKI'
-host_wiki_docker_restore_dir: '{{ bacula_dest }}{{ wiki_docker_backup_dir }}'
-
-# files restored from tape
-wiki_static_backup_file: '/mediawiki.tar'
-wiki_database_backup_file: '/wikidb.sql'
diff --git a/roles/docker-openldap/defaults/main.yml b/roles/docker-openldap/defaults/main.yml
index 58247cf..473571b 100644
--- a/roles/docker-openldap/defaults/main.yml
+++ b/roles/docker-openldap/defaults/main.yml
@@ -28,3 +28,31 @@ openldap_backup_files:
 
 # Network names
 docker_network_frontend: 'frontend'
+
+
+# the name of the image being duilt and used for the container
+openldap_image_repo: https://github.com/novatechweb/docker-openldap.git
+openldap_image_name: '{{ docker_registry_username }}/ldap'
+
+# the name of the container being started
+openldap_container_name: '{{ docker_name_prefix }}ldap'
+
+# the name of the data-volume used by the container
+openldap_cv_name: '{{ docker_name_prefix }}ldap_CV'
+openldap_dv_name: '{{ docker_name_prefix }}ldap_DV'
+
+# restore directories to temporarly store data being restored into docker containers
+openldap_docker_backup_dir: '{{ docker_backup_dir }}/LDAP'
+host_openldap_docker_restore_dir: '{{ bacula_dest }}{{ openldap_docker_backup_dir }}'
+
+
+
+# *****************************************************************************
+# Utility container - phpldapadmin
+
+# the name of the image used for the container
+phpldapadmin_image_repo: https://github.com/AndrewCooper/docker-phpldapadmin.git
+phpldapadmin_image_name: '{{ docker_registry_username }}/phpldapadmin'
+
+# the name of the container being started
+phpldapadmin_container_name: '{{ docker_name_prefix }}phpldapadmin'
diff --git a/roles/docker-openldap/tasks/main.yml b/roles/docker-openldap/tasks/main.yml
index 125fc4f..55c3e11 100644
--- a/roles/docker-openldap/tasks/main.yml
+++ b/roles/docker-openldap/tasks/main.yml
@@ -66,12 +66,12 @@
     version: master
     dest: '{{ docker_projects_dir }}/docker-openldap'
 
-
 - name: build image
   docker_image:
     name: '{{ openldap_image_name }}'
     tag: '{{ docker_image_tag }}'
     path: '{{ docker_projects_dir }}/docker-openldap'
+    force: true
 
 # *****************************************************************************
 # Create the data volumes
diff --git a/roles/docker-openldap/vars/main.yml b/roles/docker-openldap/vars/main.yml
index f66b70d..ed97d53 100644
--- a/roles/docker-openldap/vars/main.yml
+++ b/roles/docker-openldap/vars/main.yml
@@ -1,28 +1 @@
 ---
-
-# the name of the image being duilt and used for the container
-openldap_image_repo: https://github.com/novatechweb/docker-openldap.git
-openldap_image_name: '{{ docker_registry_username }}/ldap'
-
-# the name of the container being started
-openldap_container_name: '{{ docker_name_prefix }}ldap'
-
-# the name of the data-volume used by the container
-openldap_cv_name: '{{ docker_name_prefix }}ldap_CV'
-openldap_dv_name: '{{ docker_name_prefix }}ldap_DV'
-
-# restore directories to temporarly store data being restored into docker containers
-openldap_docker_backup_dir: '{{ docker_backup_dir }}/LDAP'
-host_openldap_docker_restore_dir: '{{ bacula_dest }}{{ openldap_docker_backup_dir }}'
-
-
-
-# *****************************************************************************
-# Utility container - phpldapadmin
-
-# the name of the image used for the container
-phpldapadmin_image_repo: https://github.com/AndrewCooper/docker-phpldapadmin.git
-phpldapadmin_image_name: '{{ docker_registry_username }}/phpldapadmin'
-
-# the name of the container being started
-phpldapadmin_container_name: '{{ docker_name_prefix }}phpldapadmin'
diff --git a/roles/docker-openssl/vars/main.yml b/roles/docker-openssl/defaults/main.yml
similarity index 100%
rename from roles/docker-openssl/vars/main.yml
rename to roles/docker-openssl/defaults/main.yml
diff --git a/roles/docker-openssl/tasks/main.yml b/roles/docker-openssl/tasks/main.yml
index a3828f6..95601db 100644
--- a/roles/docker-openssl/tasks/main.yml
+++ b/roles/docker-openssl/tasks/main.yml
@@ -65,6 +65,7 @@
     name: '{{ openssl_image_name }}'
     tag: '{{ docker_image_tag }}'
     path: '{{ docker_projects_dir }}/docker-openssl'
+    force: True
 
 # *****************************************************************************
 # Create the data volumes
diff --git a/roles/docker-svn/defaults/main.yml b/roles/docker-svn/defaults/main.yml
index c14b24d..2ca31ad 100644
--- a/roles/docker-svn/defaults/main.yml
+++ b/roles/docker-svn/defaults/main.yml
@@ -18,3 +18,17 @@ svn_repos: 'example'
 
 # Network names
 docker_network_frontend: 'frontend'
+
+# the name of the image being duilt and used for the container
+svn_image_repo: https://github.com/novatechweb/docker-svn.git
+svn_image_name: '{{ docker_registry_username }}/svn'
+
+# the name of the container being started
+svn_container_name: '{{ docker_name_prefix }}svn'
+
+# the name of the data-volume used by the container
+svn_dv_name: '{{ docker_name_prefix }}svn_DV'
+
+# restore directories to temporarly store data being restored into docker containers
+svn_docker_backup_dir: '{{ docker_backup_dir }}/SVN'
+host_svn_docker_restore_dir: '{{ bacula_dest }}{{ svn_docker_backup_dir }}'
diff --git a/roles/docker-svn/vars/main.yml b/roles/docker-svn/vars/main.yml
index f8e412d..ed97d53 100644
--- a/roles/docker-svn/vars/main.yml
+++ b/roles/docker-svn/vars/main.yml
@@ -1,15 +1 @@
 ---
-
-# the name of the image being duilt and used for the container
-svn_image_repo: https://github.com/novatechweb/docker-svn.git
-svn_image_name: '{{ docker_registry_username }}/svn'
-
-# the name of the container being started
-svn_container_name: '{{ docker_name_prefix }}svn'
-
-# the name of the data-volume used by the container
-svn_dv_name: '{{ docker_name_prefix }}svn_DV'
-
-# restore directories to temporarly store data being restored into docker containers
-svn_docker_backup_dir: '{{ docker_backup_dir }}/SVN'
-host_svn_docker_restore_dir: '{{ bacula_dest }}{{ svn_docker_backup_dir }}'
